_posts/en/newsletters/2018-12-18-newsletter.md

```diff
@@ -0,0 +1,157 @@
+  of the expected differences themselves.  That may not sound
+  exciting---the `rsync` tool has done that for over two decades---but
+  libminisketch allows transferring the differences *without knowing what
+  they are ahead of time.*
+
+    For example, Alice has elements 1, 2, and 3.  Bob has elements 1 and 3.
+    Despite neither knowing which elements the other has, Alice can send
+    Bob a *sketch* the size of a single element that contains enough
+    information for him reconstruct element 2.  If Bob instead has
+    elements 1 and 2 (not 3), the exact same sketch allows him to
+    reconstruct element 3.  Alternatively, if Bob sends Alice a sketch
+    based on his two-element set while Alice has her three-element
+    set, she can determine which element Bob is missing and send it
+    to him directly.
+
+    These sketches can provide a powerful new way to optimize relay of
+    unconfirmed transactions for the Bitcoin P2P network.  The current
+    gossip-based mechanism has each node receiving or sending 32-byte
+    identifiers for each transaction for each of their peers.  For
+    example, if you have 100 peers, you send or receive 3,200 bytes of
+    announcements, plus overhead, for what is (on average) just a 400
+    byte transaction.  An early estimate using a [simulator][naumenko
+    relay simulator] indicates combining sketches with shortened
+    transaction identifiers (for relay only) could reduce total
+    transaction propagation bandwidth by a factor of 44x.  Sketches also
+    have the potential to provide other desirable features---for
+    example, LN protocol developer Rusty Russel started a [thread][ln
+    minisketch] on the Lightning-Dev mailing list about using them for
+    sending LN routing table updates.
+
+- **Description about what might be included in a Schnorr/Taproot soft
+  fork:** Bitcoin protocol developer Anthony Towns has [posted][towns
+  schnorr taproot] a well-written email describing what he thinks
+  ought to be included in a soft fork that adds the Schnorr signature
+  scheme plus Taproot-style MAST to Bitcoin.  This is not a formal
+  proposal, but it's similar to opinions we've heard from other
+  developers and so should provide a good overview of current thinking.
+
+- **LN protocol IRC meeting:** protocol developers for LN have agreed to
+  try converting their periodic meeting for developing the LN
+  specification from a Google Hangout to an IRC meeting after receiving
+  requests from several developers.  The [next meeting][ln irc meeting]
+  will be Tuesday, January 8th, at 19:00 (UTC).
+
+## Notable code changes
+
+*Notable code changes this week in [Bitcoin Core][core commits],
+[LND][lnd commits], [C-lightning][cl commits], and [libsecp256k1][secp
+commits].*
+
+- [Bitcoin Core #14573][] moves various miscellaneous options that
+  opened separate dialogues in the Bitcoin-Qt GUI to a new top-level
+  menu item labeled *Window*, hopefully making those options easier to
```

Comment:
Nit: Could drop "hopefully"?

Reply:
We could, but I don't think it's a guaranteed result, so I like the extra word here too.